Kernel/Devices/HID/I8042Controller.h

```diff
@@ -81,68 +81,84 @@ class I8042Device {
 
 class PS2KeyboardDevice;
 class PS2MouseDevice;
+class HIDManagement;
 class I8042Controller : public RefCounted<I8042Controller> {
     friend class PS2KeyboardDevice;
     friend class PS2MouseDevice;
 
 public:
     static NonnullRefPtr<I8042Controller> initialize();
 
-    void detect_devices();
+    ErrorOr<void> detect_devices();
 
     bool reset_device(HIDDevice::Type device)
     {
         SpinlockLocker lock(m_lock);
-        return do_reset_device(device);
+        // FIXME: Propagate errors properly
+        if (auto result = do_reset_device(device); result.is_error())
+            return false;
+        return true;
     }
 
     u8 send_command(HIDDevice::Type device, u8 command)
     {
         SpinlockLocker lock(m_lock);
-        return do_send_command(device, command);
+        // FIXME: Propagate errors properly
+        return MUST(do_send_command(device, command));
     }
     u8 send_command(HIDDevice::Type device, u8 command, u8 data)
     {
         SpinlockLocker lock(m_lock);
-        return do_send_command(device, command, data);
+        // FIXME: Propagate errors properly
+        return MUST(do_send_command(device, command, data));
     }
 
     u8 read_from_device(HIDDevice::Type device)
     {
         SpinlockLocker lock(m_lock);
-        return do_read_from_device(device);
+        // FIXME: Propagate errors properly
+        return MUST(do_read_from_device(device));
     }
 
     void wait_then_write(u8 port, u8 data)
     {
         SpinlockLocker lock(m_lock);
-        do_wait_then_write(port, data);
+        // FIXME: Propagate errors properly
+        MUST(do_wait_then_write(port, data));
     }
 
     u8 wait_then_read(u8 port)
     {
         SpinlockLocker lock(m_lock);
-        return do_wait_then_read(port);
+        // FIXME: Propagate errors properly
+        return MUST(do_wait_then_read(port));
     }
 
-    void prepare_for_output();
-    void prepare_for_input(HIDDevice::Type);
+    ErrorOr<void> prepare_for_output();
+    ErrorOr<void> prepare_for_input(HIDDevice::Type);
 
     bool irq_process_input_buffer(HIDDevice::Type);
 
     RefPtr<MouseDevice> mouse() const;
     RefPtr<KeyboardDevice> keyboard() const;
 
+    // Note: Thes function exists only for the initialization process of the controller
```

Comment:
Typo: "thes"

Reply:
Thanks for noticing and bringing my attention to this :)